Add minimal downstream documentation usage (#118)
janus-idp/operator: .rhdh/docs/openshift.md

## Using the Operator to manage Red Hat Developer Hub (RHDH)

### Installing RHDH using Operator

**Prerequisites**

- You are logged in using the OpenShift web console.

> - You have configured the appropriate roles and permissions within your project to create an application.

Reviewer: Can we specify or add a link to the sufficient documentation for the "appropriate roles and permissions" ?

Reviewer 2: does this happen automatically when you create a project/namespace? 

For example on my clusterbot instance (logged in as `kubeadmin`), I just do this to create a project for my backstage deployment, and then add a pull secret to it so I can pull from quay.io:

```
oc new-project mybackstage
oc create -f /tmp/my_quay_pull_secret -n mybackstage
```

where the secret file is the one provided by Quay, eg.,

```
apiVersion: v1
kind: Secret
metadata:
  name: rhdh-pull-secret
data:
  .dockerconfigjson: your-digest-goes-here=
type: kubernetes.io/dockerconfigjson
```

Author: Added link in [`1886938` (#118)](https://github.com/janus-idp/operator/pull/118/commits/1886938d5c790b50f4e4209c724af26afc39ee07)

       baseUrl: https://backstage-<CUSTOM_RESOURCE_NAME>-<NAMESPACE_NAME>.<OPENSHIFT_CLUSTER_ROUTE_HOST>
       cors:
         origin: https://backstage-<CUSTOM_RESOURCE_NAME>-<NAMESPACE_NAME>.<OPENSHIFT_CLUSTER_ROUTE_HOST>
```

<details>
<summary>Example</summary>

```yaml
kind: ConfigMap
apiVersion: v1
metadata:
  name: app-config-rhdh
data:
  "app-config-rhdh.yaml": |
    app:
      title: Red Hat Developer Hub
      baseUrl: https://backstage-backstage-sample-my-ns.apps.ci-ln-vtkzr22-72292.origin-ci-int-gce.dev.rhcloud.com
    backend:
      auth:
        keys:
          - secret: "${BACKEND_SECRET}"
      baseUrl: https://backstage-backstage-sample-my-ns.apps.ci-ln-vtkzr22-72292.origin-ci-int-gce.dev.rhcloud.com
      cors:
        origin: https://backstage-backstage-sample-my-ns.apps.ci-ln-vtkzr22-72292.origin-ci-int-gce.dev.rhcloud.com
```

</details>

**Notes**:
- There is a mandatory Backend Auth Key for Backstage. This will reference an environment variable backed by an OpenShift Secret.
- You need to know the external URL of your Backstage instance and set it in the `app.baseUrl`, `backend.baseUrl` and `backend.cors.origin` fields of the application configuration.

>   By default, this will be named as follows: `https://backstage-<CUSTOM_RESOURCE_NAME>-<NAMESPACE_NAME>.<OPENSHIFT_CLUSTER_ROUTE_HOST>`,

Reviewer: Can we specify how to know these parameters, specifically OPENSHIFT_CLUSTER_ROUTE_HOST ?

Reviewer 2: what about `oc cluster-info` ?

Reviewer: I have nothing against 'oc cluster-info' or any other documented methods :)

Author: `oc cluster-info` seems to return the control plane URL, not the cluster domain URL.
For example, I had a sample Backstage Route exposed at https://backstage-bs1-my-ns.apps.586d5a08b54d03a8eeda.hypershift.aws-2.ci.openshift.org/, while `oc cluster-info` returned `https://a082bf2f1976b4869842662c61c71403-082d7fd8d45a63c6.elb.us-east-1.amazonaws.com:6443` as the control plane URL.

I found that `oc get ingresses.config/cluster -o jsonpath='{.spec.domain}'` returns the right information.
Fixed in [`2c6d22e` (#118)](https://github.com/janus-idp/operator/pull/118/commits/2c6d22e80c8bba0cbfc3c9e955bee21a533eb023)

> ## Using the Operator to manage Red Hat Developer Hub (RHDH)

Reviewer 2: Can you switch this file to asciidoc? Downstream processes require .adoc as input to Pantheon.

### Installing RHDH using Operator

**Prerequisites**

- You are logged in using the OpenShift web console.

> - You have configured the appropriate roles and permissions within your project to create an application.

Reviewer 3: In order to install the RHDP operator, the user must have a cluster admin role assigned.

Reviewer 2: added suggestion to the previous line of the prereqs.

  See the [Red Hat OpenShift documentation on Building applications](https://docs.openshift.com/container-platform/4.14/applications/index.html) for more details.

> - You (or an administrator) have added the RHDH Operator Catalog Source. See https://gitlab.cee.redhat.com/rhidp/rhdh/-/blob/rhdh-1-rhel-9/docs/RHDH.adoc#user-content-operator-installation for more details. This is required until the Operator is fully available on the OpenShift Operator Hub.

Reviewer 2: ```suggestion
- If installing a CI build from quay.io/rhdh, you (or an administrator) have [added a RHDH Operator Catalog Source](https://gist.github.com/nickboldt/d54ba9d7264a5480d14eea6c3bd2ecdf). For GA releases, this is not required. 
```

> **Procedure**

Reviewer 3: This is for the deployment of RHDH using Backstage CR, assuming the operator is already installed? If the operator installation procedure should be added too.

Reviewer 2: Agreed, you're missing the steps for selecting the operator from OperatorHub and enabling the subscription (installing the bundle). See suggestion below for adding a step 0 to your list.

1. Create a project in the OpenShift, if not present.
   For more information about creating a project in OpenShift, see the [Red Hat OpenShift documentation](https://docs.openshift.com/container-platform/4.14/applications/projects/working-with-projects.html#creating-a-project-using-the-web-console_projects).

> 2. Switch to **Developer** mode on your Red Hat OpenShift web console.

Reviewer 2: ```suggestion
2. Switch to the **Developer** view in your Red Hat OpenShift web console. 
```
